### PR TITLE
feat(utxo-coredao): descriptor for staking output

### PR DESCRIPTION
**Pull request by davidkaplanbitgo**
TICKET: BTC-1578

### CODE REVIEW

File: modules/utxo-coredao/src/descriptor.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  if (m > orderedKeys.length || m < 1) {
+    throw new Error(
+      `m (${m}) must be less than or equal to the number of keys (${orderedKeys.length}) and greater than 0`
+    );
+  }
+  if (locktime <= 0) {
+    throw new Error(`locktime (${locktime}) must be greater than 0`);
+  }
+
+  const xpubs = orderedKeys.map((key) => key.toBase58() + '/*');
```
Comment (davidkaplanbitgo) [inline modules/utxo-coredao/src/descriptor.ts:30]: should we default to neutering these keys unless explicitly said? Would protect

File: modules/utxo-coredao/test/unit/utils.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { promises as fs } from 'fs';
+import { Descriptor, Psbt } from '@bitgo/wasm-miniscript';
+import * as utxolib from '@bitgo/utxo-lib';
+
+export async function getFixtureString(path: string, defaultValue: string): Promise<string> {
```
Comment (OttoAllmendinger) [inline modules/utxo-coredao/test/unit/utils.ts:5]: little quality of life improvements

I should factor this into a library some day...

```suggestion
export async function getFixture(path: string, defaultValue: unknown): Promise<unknown> {
```

File: modules/utxo-coredao/test/unit/utils.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  try {
+    return await fs.readFile(path, 'utf8');
+  } catch (e) {
+    if ((e as any).code === 'ENOENT') {
+      await fs.writeFile(path, defaultValue, 'utf8');
```
Comment (OttoAllmendinger) [inline modules/utxo-coredao/test/unit/utils.ts:10]:
```suggestion
      await fs.writeFile(path, encode(path, defaultValue), 'utf8');
```

```
function encode(path: string, defaultValue: unknown): string {
  if (path.endsWith('.txt')) {
    return String(v);
  }
  if (path.endsWith('.json')) {
   return JSON.stringify(v, null, 2);
  }
  throw new Error(`unrecognized path ${path}`);
}

function decode(path, v: string): unknown {
  if (path.endsWith('.txt')) {
   return v;
  }
  if (path.endsWith('.json')) {
    return JSON.parse(v);
  }
  throw new Error(`unrecognized path ${path}`);
}
```

File: modules/utxo-coredao/test/unit/utils.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { promises as fs } from 'fs';
+import { Descriptor, Psbt } from '@bitgo/wasm-miniscript';
+import * as utxolib from '@bitgo/utxo-lib';
+
+export async function getFixtureString(path: string, defaultValue: string): Promise<string> {
+  try {
+    return await fs.readFile(path, 'utf8');
```
Comment (OttoAllmendinger) [inline modules/utxo-coredao/test/unit/utils.ts:7]:
```suggestion
    return decode(path, await fs.readFile(path, 'utf8'));
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
small tweaks to `getFixture`

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/utxo-coredao/src/descriptor.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/utxo-coredao/src/descriptor.ts:30]: sounds good

File: modules/utxo-coredao/src/descriptor.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * Create a wallet descriptor to produce a coredao staking address for a standard BitGo wallet
+ * @param scriptType
+ * @param locktime
+ * @param rootWalletKeys
+ */
+export function createWalletDescriptor(
+  scriptType: ScriptType,
+  locktime: number,
+  rootWalletKeys: bitgo.RootWalletKeys
```
Comment (OttoAllmendinger) [inline modules/utxo-coredao/src/descriptor.ts:44]:
```suggestion
  walletKeys: bitgo.Triple<BIP32Interface>
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
rootWalletKeys is a bit fixedScript specific

**Review by OttoAllmendinger — APPROVED** [review]

**Pull request merged**